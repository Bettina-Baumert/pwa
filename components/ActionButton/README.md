ActionButton Component
---

The ActionButton serves as a clickable Button and can also show a loading indicator based on the loading prop.

## Getting Started

```
import { ActionButton } from 'Templates/components';

<ActionButton onClick={handleClick} />
```

## Props

### onClick (required)

_Type_: `Function`  

Accepts a callback function for the components click event.

### loading

_Type_: `boolean`  
_Default_: `false`  

Shows a loading indicator if set to true.

###### Usage:

```
<ActionButton loading={true} ... />
```
---
