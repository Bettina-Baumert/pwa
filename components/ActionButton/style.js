/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import cxs from 'cxs';
import { gap } from 'Styles/variables';

const halfGapBig = gap.big / 2;

const containerBase = {
  textAlign: 'center',
};

const container = cxs({
  ...containerBase,
  margin: `${halfGapBig}px 0`,
});

const containerCircle = cxs({
  ...containerBase,
  margin: `${halfGapBig + 4}px 0`,
});

export default {
  container,
  containerCircle,
};
