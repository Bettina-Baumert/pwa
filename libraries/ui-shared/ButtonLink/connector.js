import { connect } from 'react-redux';
import { historyPush } from '@shopgate/pwa-common/actions/router';

/**
 * @param {Function} dispatch The redux dispatch function.
 * @return {Object}
 */
const mapDispatchToProps = dispatch => ({
  navigate: location => dispatch(historyPush({ pathname: location })),
});

export default connect(null, mapDispatchToProps);
