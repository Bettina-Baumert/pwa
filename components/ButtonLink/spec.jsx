/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import React from 'react';
import { mount } from 'enzyme';
import ParsedMockLink from '@shopgate/pwa-common/components/Router/helpers/parsed-link/index.mock';
import ActionButton from '../ActionButton';
import ButtonLink from './index';

const mockedParsedLink = new ParsedMockLink('http://example.com');
jest.mock(
  '@shopgate/pwa-common/components/Router/helpers/parsed-link',
  () => () => mockedParsedLink
);

describe('<ButtonLink>', () => {
  describe('On click action', () => {
    const originalActionButtonClickDelay = ActionButton.clickDelay;
    beforeAll(() => {
      ActionButton.clickDelay = 0;
    });
    afterAll(() => {
      ActionButton.clickDelay = originalActionButtonClickDelay;
    });
    it('should create component and open page on click', (done) => {
      const component = mount(
        <ButtonLink href="https://example.com">
          Text inside
        </ButtonLink>
      );
      expect(component).toMatchSnapshot();
      expect(mockedParsedLink.openFunctionMock.mock.calls.length).toBe(0);
      component.find('RippleButton').simulate('click');
      component.update();
      setTimeout(() => {
        expect(mockedParsedLink.openFunctionMock.mock.calls.length).toBe(1);
        done();
      }, ActionButton.clickDelay + 1);
    });
  });
});
