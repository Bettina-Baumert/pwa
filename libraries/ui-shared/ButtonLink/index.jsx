import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ActionButton from '../ActionButton';
import connect from './connector';

/**
 * Simple wrapper around ActionButton so it's easy to render buttons which behave as links.
 */
class ButtonLink extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
    href: PropTypes.string.isRequired,
    navigate: PropTypes.func,
    noGap: PropTypes.bool,
  };

  static defaultProps = {
    navigate: () => {},
    noGap: false,
  };

  /**
   * Opens the given url on click.
   * @returns {undefined}
   */
  handleClick = () => {
    this.props.navigate(this.props.href);
  };

  /**
   * Renders an ActionButton and handles link handling.
   * @returns {XML}
   */
  render() {
    return (
      <ActionButton onClick={this.handleClick} type="secondary" noGap={this.props.noGap}>
        {this.props.children}
      </ActionButton>
    );
  }
}

export default connect(ButtonLink);
