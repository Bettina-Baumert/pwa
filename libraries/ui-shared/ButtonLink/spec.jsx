import React from 'react';
import { shallow } from 'enzyme';
import ActionButton from '../ActionButton';
import ButtonLink from './index';

describe.skip('<ButtonLink>', () => {
  describe('On click action', () => {
    beforeAll(() => {
      ActionButton.clickDelay = 0;
    });
    it('should create component and open page on click', () => {
      const component = shallow((
        <ButtonLink href="https://example.com">Text inside</ButtonLink>
      ));
      expect(component).toMatchSnapshot();
    });
  });
});
