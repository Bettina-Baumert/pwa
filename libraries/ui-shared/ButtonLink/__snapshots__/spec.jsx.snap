// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonLink> On click action should create component and open page on click 1`] = `
<ActionButton
  className=""
  disabled={false}
  flat={true}
  loading={false}
  noGap={false}
  onClick={[Function]}
  rippleClassName=""
  rippleSize={null}
  testId={null}
  type="secondary"
  wrapContent={true}
>
  Text inside
</ActionButton>
`;
